blaze/compute/sql.py

```diff
@@ -176,6 +176,12 @@ def compute_up(t, s, **kwargs):
     return getattr(t, 'op', getattr(safuncs, sym, getattr(sa.func, sym)))(s)
 
 
+@dispatch(Selection, sa.sql.ColumnElement)
+def compute_up(expr, data, scope=None, **kwargs):
+    predicate = compute(expr.predicate, data, post_compute=False)
+    return sa.select([data]).where(predicate)
```

Comment:
I think in all cases I know the kind of thing I'm getting back (`Select`, `Table`, `ColumnElement`, etc)